types/csv2json/index.d.ts

```diff
@@ -0,0 +1,15 @@
+// Type definitions for csv2json 1.4
+// Project: https://github.com/julien-f/csv2json
+// Definitions by: Piotr Roszatycki <https://github.com/dex4er>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Writable } from "stream";
+
+export interface CSV2JSONOptions {
+  dynamicTyping?: boolean;
+  separator?: string;
+}
+
+export default function csv2json(options: CSV2JSONOptions): Writable;
```

Comment:
You have an `export default` here but there isn't one in the actual implementation of `csv2json`.
Please follow the following template (i.e., merging a namespace containing types you wish to export with the function you want to be callable).

Reply:
So what about `"esModuleInterop": true`? This `export default` works if I use this setting. If new `import` syntax is forbidden then why `dtslint` forces me to enable `esModuleInterop` setting?

Reply:
I'm not sure I understand your concern; the `import` syntax you were using is fine when `esModuleInterop` is enabled.

Reply:
My typings for other modules were rejected because of this `export default`. I would agree that I can use old-styled `export = foo` and namespaces to get the compatibility with older TypeScript in case if someone don't want to migrate to the newer version.

Reply:
Hey @dex4er, `export =` is not just an old style - it's a faithful representation of how the module was actually authored. Depending on the future direction of ECMAScript module interop along with formats like CommonJS, it makes it dramatically easier for consumers (and our team) to reason about how to best consume a module if its `.d.ts` file accurately describes the original format.